index.js

```diff
@@ -0,0 +1,183 @@
+    }
+
+    if (typeof this._watch !== 'boolean') {
+      throw new Error('`options.watch` is of an invalid type.');
+    }
+
+    if (typeof this._taskDependencies !== 'object') {
+      throw new Error('`options.taskDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._localDependencies !== 'object') {
+      throw new Error('`options.localDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._remoteDependencies !== 'object') {
+      throw new Error('`options.remoteDependencies` is of an invalid type.');
+    }
+
+    if (typeof this._bundledSpecPath !== 'string') {
+      throw new Error('`options.bundledSpecPath` is of an invalid type.');
+    }
+ 
+    if (typeof this._port !== 'number') {
+      throw new Error('`options.karmaConfig.port` is of an invalid type.');
+    }
+ 
+    if (typeof this._bundlePath !== 'string') {
+      throw new Error('`options.bundlePath` is of an invalid type.');
+    }
+   }
+
+  _registerTasks() {
+    this._registerFetchRemoteDeps();
+    this._registerBundleDeps();
+    this._registerCombine();
+    this._registerRunSpec();
+    this._registerErik();
+  }
+
+  _registerFetchRemoteDeps() {
+    this._gulp.task('erik-fetch-remote-deps', () => {
+      return streamqueue({
+        objectMode: true
+      }, ...this._remoteDependencies.map((urlStr) => {
+        const url = URL.parse(urlStr);
+        const base = `${url.protocol}//${url.host}`;
+        const path = url.path;
+
+        return remoteSrc([path], {
+          base
+        });
+      }))
+        .pipe(concat('remote-deps.js'))
+        .pipe(this._gulp.dest(this._erikPath));
+    });
+  }
+
+  _registerBundleDeps() {
+    this._gulp.task('erik-bundle-deps', () => {
+      return this._gulp.src([
+        'erik/remote-deps.js',
+        ...this._localDependencies
+      ])
+        .pipe(concat('bundled-deps.js'))
+        .pipe(this._gulp.dest(this._erikPath));
+    });
+  }
+
+  _registerCombine() {
+    /**
+     * Combine the bundled dependencies and specs to ensure that the dependencies are available to
+     * the test suite.
+     */
+    this._gulp.task('erik-combine', () => {
+      return this._gulp.src([
+        'erik/bundled-deps.js',
+        this._bundledSpecPath
+      ])
+        .pipe(concat('combined.js'))
+        .pipe(this._gulp.dest(this._erikPath));
+    });
+  }
+
+  _registerRunSpec() {
+    this._gulp.task('erik-run-spec', (done) => {
+      new karmaServer.start({
+        frameworks: ['jasmine'],
+        browsers: ['PhantomJS'],
+        reporters: ['mocha'],
+
+        /**
+         * Pass an empty object for `preprocessors` in order to disable Karma's default processing
+         * of coffescript files.
+         */
+        preprocessors: {
+        },
+
+        files: [
+          'erik/combined.js'
+        ],
+
+        port: this._port,
+
+        autoWatch: this._watch,
+        singleRun: !this._watch
+      }, function(error) {
```

Comment:
I've updated this handling to be more descriptive. The argument here is now `exitCode` instead of `error`. This callback is only every called once. In the case of `!singleRun`, it will be run when the user ends the `gulp erik` process.